test/invite.js

```diff
@@ -14,59 +14,62 @@ var createSbot = require('../')
   .use(require('ssb-friends'))
   .use(require('ssb-ws'))
 
-function all(stream, cb) {
+function all (stream, cb) {
   return pull(stream, pull.collect(cb))
 }
 
 var wsConnections = {
   incoming: {
-    net: [{ scope: ["local", "device"], "transform": "shs", host: "::" }],
-    ws: [{ scope: ["local", "device"], "transform": "shs", host: "::" }]
+    net: [{ scope: ['local', 'device'], 'transform': 'shs', host: '::' }],
+    ws: [{ scope: ['local', 'device'], 'transform': 'shs', host: '::' }]
   },
   outgoing: {
-    net: [{ transform: "shs" }],
-    ws: [{ transform: "shs" }]
+    net: [{ transform: 'shs' }],
+    ws: [{ transform: 'shs' }]
   }
 }
 
 tape('test invite.accept api', function (t) {
   var alice = createSbot({
-    temp: 'test-invite-alice2', timeout: 100,
+    temp: 'test-invite-alice2',
+    timeout: 100,
     allowPrivate: true,
-    keys: ssbKeys.generate(),
+    keys: ssbKeys.generate()
   })
 
   var bob = createSbot({
-    temp: 'test-invite-bob2', timeout: 100,
-    keys: ssbKeys.generate(),
+    temp: 'test-invite-bob2',
+    timeout: 100,
+    keys: ssbKeys.generate()
   })
 
   var carol = createSbot({
-    temp: 'test-invite-carol2', timeout: 100,
-    keys: ssbKeys.generate(),
+    temp: 'test-invite-carol2',
+    timeout: 100,
+    keys: ssbKeys.generate()
   })
 
-  if(!alice.getAddress('device'))
-    throw new Error('alice must have device address')
-  if(!alice.getAddress('local'))
-    throw new Error('alice must have local address')
+  if (!alice.getAddress('device')) { throw new Error('alice must have device address') }
+  if (!alice.getAddress('local')) { throw new Error('alice must have local address') }
 
-  //request a secret that with particular permissions.
+  // request a secret that with particular permissions.
 
   alice.invite.create(1, function (err, invite) {
-    if(err) throw err
-    //test that invite is accepted with quotes around it.
+    if (err) throw err
+    // test that invite is accepted with quotes around it.
     bob.invite.accept(JSON.stringify(invite), function (err) {
-      if(err) throw err
+      if (err) throw err
       alice.friends.hops({
         source: alice.id, dest: bob.id
       }, function (err, hops) {
-        if(err) throw err
+        if (err) throw err
         t.equal(hops[bob.id], 1, 'alice follows bob')
         carol.invite.accept(invite, function (err) {
+          if (err) throw err
```

Comment:
@dominictarr This is the last error being thrown, otherwise tests pass (:tada:). I'm under the impression that Alice created a single-use invite which Bob used, so Carol's invite *should* fail. This should be replaced with `t.ok(err)`, right?

```
/home/christianbundy/src/ssbc/scuttlebot/test/invite.js:68
          if (err) throw err
                   ^

Error: invite not accepted
    at /home/christianbundy/src/ssbc/scuttlebot/plugins/invite.js:223:32
    at Object._requests.(anonymous function) (/home/christianbundy/src/ssbc/scuttlebot/node_modules/packet-stream/index.js:38:5)
    at PacketStream._onrequest (/home/christianbundy/src/ssbc/scuttlebot/node_modules/packet-stream/index.js:151:26)
  Error: method:invite,use is not in list of allowed methods
    at Function.perms.pre (/home/christianbundy/src/ssbc/scuttlebot/node_modules/muxrpc/permissions.js:88:14)
    at Object.<anonymous> (/home/christianbundy/src/ssbc/scuttlebot/node_modules/muxrpc/local-api.js:35:21)
    at Object.request (/home/christianbundy/src/ssbc/scuttlebot/node_modules/muxrpc/stream.js:46:17)
    at PacketStream._onrequest (/home/christianbundy/src/ssbc/scuttlebot/node_modules/packet-stream/index.js:161:17)
    at PacketStream.write (/home/christianbundy/src/ssbc/scuttlebot/node_modules/packet-stream/index.js:134:41)
    at /home/christianbundy/src/ssbc/scuttlebot/node_modules/muxrpc/pull-weird.js:56:15
    at /home/christianbundy/src/ssbc/scuttlebot/node_modules/pull-stream/sinks/drain.js:24:37
    at /home/christianbundy/src/ssbc/scuttlebot/node_modules/pull-goodbye/node_modules/pull-stream/throughs/filter.js:17:11
    at Object.cb (/home/christianbundy/src/ssbc/scuttlebot/node_modules/packet-stream-codec/index.js:111:11)
    at drain (/home/christianbundy/src/ssbc/scuttlebot/node_modules/pull-reader/index.js:39:14)
```

Reply:
yes, the next thing is it checks that alice didn't follow carol. `t.ok(err)` is good.